src/legacy/core_plugins/timelion/public/vis/timelion_request_handler.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// @ts-ignore
+import { buildEsQuery, getEsQueryConfig, Filter } from '@kbn/es-query';
+// @ts-ignore
+import { timezoneProvider } from 'ui/vis/lib/timezone';
+import { KIBANA_CONTEXT_NAME } from 'src/plugins/expressions/public';
+import { Query } from 'src/legacy/core_plugins/data/public';
+import { TimeRange } from 'src/plugins/data/public';
+import { VisParams } from 'ui/vis';
+import { toastNotifications } from 'ui/notify';
+import { i18n } from '@kbn/i18n';
+import { TimelionSetupDependencies } from '../plugin';
+
+interface Stats {
+  cacheCount: number;
+  invokeTime: number;
+  queryCount: number;
+  queryTime: number;
+  sheetTime: number;
+}
+
+interface Sheet {
+  list: Array<Record<string, unknown>>;
+  render: Record<string, unknown>;
+  type: string;
+}
+
+export interface TimelionSuccessResponse {
+  sheet: Sheet[];
+  stats: Stats;
+  visType: string;
+  type: KIBANA_CONTEXT_NAME;
+}
+
+export function getTimelionRequestHandler(dependencies: TimelionSetupDependencies) {
```

Comment:
nit: i would directly destructure this and not use TimelionSetupDependencies type at all

Reply:
`TimelionSetupDependencies` has been removed and replaced by `TimelionVisualizationDependencies`.

But if we do destructuring we also need to define types of its fields.

It may seem like:

```
getTimelionRequestHandler({ uiSettings, http }: { uiSettings: Pick<TimelionVisualizationDependencies, 'uiSettings'>, http: Pick<TimelionVisualizationDependencies, 'http'> })
```

It's slight huge and ugly I guess, so I left as 
```
getTimelionRequestHandler(dependencies: TimelionVisualizationDependencies)
```